assets/src/edit-story/components/statusCheck/statusCheck.js

```diff
@@ -17,51 +17,27 @@
 /**
  * External dependencies
  */
-import { useCallback, useEffect } from 'react';
-import { useFeature } from 'flagged';
-
-/**
- * WordPress dependencies
- */
-import { __ } from '@wordpress/i18n';
+import { useEffect, useCallback, useState } from 'react';
 
 /**
  * Internal dependencies
  */
 import { useAPI } from '../../app';
 import { getContent } from './utils';
+import StatusCheckFailed from './statusCheckFailed';
 
 function StatusCheck() {
   const {
     actions: { getStatusCheck },
   } = useAPI();
-  const statusCheckEnabled = useFeature('statusCheck');
-
-  const doStatusCheck = useCallback(() => {
-    if (!statusCheckEnabled) {
-      return;
-    }
-
-    async function performRequests() {
-      const content = getContent();
-
-      try {
-        await getStatusCheck(content);
-        // eslint-disable-next-line no-console
-        console.log(__('Status Check successful.', 'web-stories'));
-      } catch (err) {
-        // eslint-disable-next-line no-console
-        console.log(__('Status Check failure.', 'web-stories'));
-      }
-      // eslint-disable-next-line no-console
-      console.log(__('Status Check finished.', 'web-stories'));
-    }
-
-    performRequests();
-  }, [getStatusCheck, statusCheckEnabled]);
+  const [showDialog, setShowDialog] = useState(false);
+  const closeDialog = useCallback(() => setShowDialog(false), []);
 
-  useEffect(doStatusCheck);
+  useEffect(() => {
+    // If it succeeds, do nothing. Only in case of failure do we want to alert the user.
+    getStatusCheck(getContent()).catch(() => setShowDialog(true));
+  }, [getStatusCheck]);
```

Comment:
Is it intended for the status check is run just once -- later, even if the save fails, the status check doesn't actually run/display modal. Should this perhaps run again if auto-saving fails for example? Thoughts?

Reply:
For now this is meant as a status check when loading the editor for the first time, indicating to the user that it might not work, to prevent any data loss.

Dealing with failed autosaves can be handled in #4840.

See also #4839, #1033

Reply:
Alright, sounds good. Asked since the wording can be a bit confusing in this case -- e.g. I open a new story and right away it lets me know that "Unable to save your story" -- as if I had tried saving it.